apps/go-mod-validator/test/e2e.test.ts

```diff
@@ -0,0 +1,81 @@
+import { describe, it, expect } from "vitest";
+import { run } from "../src/go-mod-validator";
+import { join } from "path";
+
+const testDataDir = "./test/data";
+
+function setup(repoName: string) {
+  process.env["INPUT_GO-MOD-DIR"] = join(testDataDir, repoName);
+  process.env["INPUT_GITHUB-TOKEN"] = process.env.GITHUB_TOKEN;
+  process.env["INPUT_DEP-PREFIX"] = "github.com/smartcontractkit";
+  process.env["GITHUB_STEP_SUMMARY"] = "/dev/null";
+}
+
+describe("e2e tests", () => {
+  it("chainlink - should match snapshot", { timeout: 20_000 }, async () => {
+    setup("chainlink");
+    const summary = await run();
+
+    expect(summary).toMatchInlineSnapshot(`
```

Comment:
I wouldn't fully match the contents. Any update to the summary text requires an update to the test. This is more annoying than helpful.

Maybe `run` should return an object that has some boolean indicating whether validation passed, and you can use that for validation rather than strings that may change.

Reply:
This text is going to be quite static once i make use of the annotation API. It's easy to update snapshots as its automated via the `-u` flag.